{% load staticfiles %}
<html>
<head>
 <meta charset="utf-8">
<title>atSNP Homepage</title>
    <link href="{% static 'ss_viewer/css/bootstrap.min.css' %}" rel="stylesheet">
    <link href="{% static 'ss_viewer/custom.css' %}" rel="stylesheet">
    <script type="text/javascript"  src="{% static 'ss_viewer/jquery-2.2.4.min.js' %}"></script>
    <script src="{% static 'ss_viewer/js/bootstrap.min.js' %}"></script>
    <script src="{% static 'ss_viewer/google_analytics.js' %}"></script>

    <!-- does adding a link to the bootstrap JS make glyphicons appear? -->
</head>
<!-- <body  style="background-color: #DCDCDC;" >-->
<body>
<!--<nav class="navbar navbar-default navbar-fixed-top">-->
<div class="container">
  <div id="content">
      {% include "ss_viewer/navbar.html" with active='about' %}  

<!-- todo; remove in-line styles from homepage -->
<div style="box-shadow: 1px 1px 1px 1px #888888;
            padding: 1em;
            margin-bottom: 1em;">
     <!-- styles the image to resize based on the size of the window-->
     <img height="230" 
          src="{% static 'ss_viewer/revised-logo-cut.png' %}"
          style="margin-bottom: 3px;
          max-width: 100%;
            height: auto;
            width: auto;"  />
<p>
    @SNP DB provides statistical evaluation of impact of SNPs on transcription
    factor-DNA interactions based on position weight matrices (PWM).
    The current version is built on SNPs from dbSNP Build 144 for the
    human genome assembly GRCh38/hg38. atSNP DB uses Halic 
    calculations based on the statistical method in Zuo et al.,
    Bioinformatics (2015).
    atSNP first-time users can start scanning vast amounts of
    SNP-PWM combinations for gain and loss of function with two key features:
</p>          

<ul style="list-style-type: circle;">
<li> p-values quantifying the impact of SNP on the PWM matches (p-value SNP impact)</li>
<li> composite logo plots.</li>
</ul>
<p>
   @SNP identifies and quantifies best DNA sequence matches to the transcription
   factor PWMs with both the reference and the SNP alleles in a small
   window around the SNP location (up to +/- 30 base pairs and considering
   subsequences spanning the SNP position). It evaluates statistical
   significance of scores with each allele and calculates statistical
   significance of the score difference between the best matches with the reference and SNP alleles.
</p>
<p>
  The discovery of best matches with the reference and SNP alleles are
   conducted separately. As a result, genomic subsequences providing the best match
   with each allele may be different. The change in goodness of match with the reference and SNP alleles 
   alleles indicates impact of SNPs on transcription factor binding ability
   to DNA. @SNP also provides a composite logo plot for easy visualization
   of the quality of the reference and SNP allele matches to the PWM and the
   influence of SNPs on the change in match. Read more under "Help".
</p>          
</div>

<div style="box-shadow: 1px 1px 1px 1px #888888;
            padding: 1em;
            margin-bottom:1em;"> 
<p>
   <span style="font-weight:bold; font-size: 1.2em;" >News and Updates</span>
</p>
<p>
   @SNP DB human version 1.0.0 has been released on July 20, 2016.
</p>
<p>
  The current version of @SNP evaluated 133 million SNPs from the
  <a href="http://www.ncbi.nlm.nih.gov/SNP/">NCBI database</a>
  of genetic variation (dbSNP Build 144 for human genome GRCh38)
  with 2,270 PWMs (position weight matrices) from two major sources  
  of transcription factor position weight matrix libraries. These are the 
  <a href="http://jaspar.genereg.net/">JASPAR</a> library, and the
  <a href="http://compbio.mit.edu/encode-motifs/"> ENCODE</a> library.
  @SNP DB contains statistical evaluations for a total of 308 billion SNP-PWM pairs.
  The current release enables querying of three evaluations which exhibit a significant
  OWIE(?2) change and a significant match with either the reference or the SNP allele.
  Of these, results for SNP-PWM combinations with  P-value SNP impact, P-value reference, and/or
  P-value SNP less than or equal to 0.05 are currently available to query. 
</p>
</div>


<div style="box-shadow: 1px 1px 1px 1px #888888;
            padding: 1em;
            margin-bottom: 1em;">
  <p><span style="font-weight:bold; font-size: 1.2em;">Contact</span></p>
  <p>
     If you have any questions concerning atNSP, please post your question publicly 
     in the <a href="https://groups.google.com/forum/#!forum/atsnp-users">User Group</a>.
     For other questions or concerns you may send email to keles@stat.wisc.edu.
  </p>
</div>

<div style="box-shadow: 1px 1px 1px 1px #888888;
            padding: 1em;
            margin-bottom: 1em;">
  <p><span style="font-weight:bold; font-size: 1.2em;">Support</span></p>
  <p>This site is supported by: </p>

    <div style="box-shadow: 1px 1px 1px 1px #888888;
                padding: 1em;
                margin-bottom: 1em;">
         <img style="height: 40px;"
            src="{% static 'ss_viewer/CPCP-logo.png' %}" />
         <span> Center for Predective Computational Phenotyping</span>
    </div>

    <div style="box-shadow: 1px 1px 1px 1px #888888;
                padding: 1em;
                margin-bottom: 1em;">
         <!-- altering the margin-right to line up labels -->
         <img style="height: 50px; margin-right:5px"
            src="{% static 'ss_viewer/UW-logo.png' %}" />
         <span> University of Wisconsin - Madison</span>
    </div>

    <div style="box-shadow: 1px 1px 1px 1px #888888;
                padding: 1em;
                margin-bottom: 1em;">
       <div class="row">
           <div class="col-md-1">
           <img style="height: 65px;"
                src="{% static 'ss_viewer/NIH-logo.png' %}" />
           </div>
           <div class="col-md-6">
               <p style="margin-top: 0.8em; margin-left: 0.4em;">
                   National Institutes of Health - Big Data to Knowledge (BD2K) 
                   </br> Initiative under Award Number U54 AI117924
               </p>
           </div>
       </div> <!-- end the 'row' div for organizing contents.. -->
    </div>

</div>

<!-- start debug stuff -->
{% if flavor %}
   <link href="{% static 'ss_viewer/colors/' %}{{ flavor }}.css" rel="stylesheet">
{% endif %}    
<!-- end debug stuff -->


   </div>  <!-- end content div -->
</div> <!-- end container div -->
<!-- </nav> -->
</body>
</html>

