{% load staticfiles %}
<html>
<head>
 <meta charset="utf-8">
<title>atSNP FAQ </title>
    <link href="{% static 'ss_viewer/css/bootstrap.min.css' %}" rel="stylesheet">
    <link href="{% static 'ss_viewer/custom.css' %}" rel="stylesheet">
    <script type="text/javascript" src="{% static 'ss_viewer/google_analytics.js' %}"></script>
<style type="text/css">
  hgroup {
    font-weight: bold;
    color: blue;
  }
  body { font-size: 16px; }
</style>
</head>

<body>
<div class="container">
  <div id="content">
      {% include "ss_viewer/navbar.html" with active='faq' %}
<h3>FAQ</h3>


<p>
<section>
<b>Q1. Is atSNP DB exclusively for human?</b>
<ul>
<li>A1. Our current release only has data from the human genome. We plan to extend the database to mouse in the near future.</li>
</ul>
</section>
</p>



<p>
<section>
<b>Q2. Which db SNP version was used for atSNP db?</b>
<ul>
<li>A2. atSNP DB is based on dbSNP build 144 for Human Build 38 (6Rch38). Summary on the dbSNP build 144 can be found <a href="http://www.ncbi.nlm.nih.gov/SNP/snp_summary.cgi?view+summary=view+summary&build_id=144">here</a>.</li>
</ul>
</section>
</p>

<p>
<section>
<b>Q3. How does atSNP DB handle SNPs with indels, and others?</b>
<ul>
<li>A3. dbSNP classifies genetic variation in eight types. Click <a href="http://www.ncbi.nlm.nih.gov/books/NBK44447"> here </a> for the types of variation. The current atSNP DB only considers the major single nucleotide variation class; the class TAKES 88.8% of all genetic variation, specifically 132,946,852 SNPs out of 149,735,377.</li>
</ul>
</section>
</p>

<p>
<section>
<b>Q4. How are multi-allelic SNPs evaluated?</b>
<ul>
<li>A4. atSNP DB considers all possible alleles one at a time and provides atSNP statistics and p-values.</li>
</ul>
</section>
</p>

<p>
<section>
<b>Q5. Which motif libraries were used for atSNP?</b>
<ul>
<li>A5. We used JASPAR vertebrate library and known and discovered motifs for the ENCODE TF ChIP-seq datasets, which are available <a href="http://jaspar2014.genereg.net/">here for JASPAR</a>, and <a href="http://compbio.mit.edu/encode-motifs/">here for ENCODE</a>, respectively. For users who want to learn more about the libraries, visit the websites or click <a href="{% url 'ss_viewer:help-page' %}#part-3:motifTable">here</a></li>
</ul>
</section>
</p>

<p>
<section>
<b>Q6. Can I have summary statistics of atSNP DB?</b>
<ul>
<li>A6. We investigated 132,946,852 SNPs and considered 2,270 PWMs. We have evaluated total 307,602,574,340 SNP-PWM combinations.</li>
</ul>
</section>
</p>


<p>
<section>
<b>Q7. Are downloads available?</b>
<ul>
<li>A7. The current release allows downloading the results up to the first 5,000 rows in CSV format.</li>
</ul>
</section>
</p>

<p>
<section>
<b>Q8. How can I interpret the logo plots?</b>
<ul>
<li>A8. The plots are called composite logo plots and they aim to provide a quick visualization of the region around the SNP and the sequence matches to the PWM.  
    Please click <a href="{% url 'ss_viewer:help-page' %}#part-4:part-4">here</a> for the interpretation.</li>
</ul>
</section>
</p>


<p>
<section>
<b>Q9. Why can't I page past 10,000 search results?</b>
<ul>
<li>
   A9. If more than 10,000 results are returned, only the first 10,000 will be avaiable to page through. 
       To access results past the 10,000th, adjust the sort order and/or priority.
</li>
</ul>
</section>
</p>


<p>
<section>
<b>Q10. How do I cite atSNP DB?</b>
<ul>
<li>A10. 
      <ul>
      <li>
      
      1. Zuo, Chandler, Shin, Sunyoung, and Keleş, Sündüz. atSNP: transcription factor binding affinity testing for regulatory SNP detection. Bioinformatics (2015): 31, 3353-3355.</li>
      <li>
     
     2. Shin, Sunyoung, Hudson, Rebecca, Harrison, Christopher, Craven, Mark, and Keleş, Sündüz. atSNP DB: a web resource for quantifying impact of dbSNP SNPs on transcription factor binding. (2018). In preparation.
     
      </li>
     </ul>
</li>
</ul>
</section>
</p>

<p>
<section>
<b>Q11. What web browsers are supported by atSNP?</b>
<ul>
<li> A11. Everything except Internet Explorer (newest versions are called 'Edge'). </li>
</ul>
</section>
</p>

<p>
  <section>
    <b>Q12. What is a non-technical interpretation of the three p-values?</b>
    <ul>
       <li> A12-1. P-value SNP Impact: Is  the SNP a candidate for potential gain- or loss-of -function? </li>
       <li> A12-2. P-value Reference: Is there a candidate binding site at the SNP location with the reference allele?</li>
       <li> A12-3. P-value SNP:  Is there a candidate binding site at the SNP location with the SNP allele?</li>
    </ul>
  </section>
</p>

<p>
  <section>
    <b> Q13. How does atSNP DB define gain-of-function?</b>
    <ul> 
    <li>
    A13. atSNP DB generates candidates for gain of function by identifying (SNP, PWM) pairs with
    <p style="margin-left: 2em;"> 
    P-value SNP Impact &lt;= 0.05 <br />
    P-value Reference &gt; 0.05   <br /> 
    P-value SNP &lt;= 0.05        <br />
    </p>
    
    This essentially identifies (SNP, PWM) pairs which do not have a statistically significant match with the reference allele and the SNP allele leads to a statistically significant match (i.e., creating a binding site).
    
    Note  that these p-values are only suggestive and users should make them more stringent. 
   </li>
  </section>
</p>

<p>
   <section>
     <b>Q14. How does atSNP DB define loss-of-function?</b>
     <ul> 
        <li>
        A14. atSNP DB generates candidates for loss-of-function by identifying (SNP, PWM) pairs  with
       <p style="margin-left: 2em;"> 
         P-value SNP Impact &lt;= 0.05<br />
         P-value Reference &lt;= 0.05 <br />
         P-value SNP &gt; 0.05        <br />
       </p>
         This essentially identifies (SNP, PWM) pairs which have a statistically significant match with the reference allele and the SNP allele lead to a  statistically insignificant match (i.e., destroying a binding site).
         Note  that these p-values are only suggestive and users should make them more stringent. 
       </li>
     </ul>
   </section>
</p>

<p>
<section>
    <b>Q15. How are motifs (PWMs) classified into low, moderate, high, and very high degeneracy groups?</b>
    <ul>
         <li>A15. This is based on computing information content of each individual position and then averaging the information content across all positions.  Then, each PWM is classified as follows based on its average information content.
         </li>
         <li>
         <!-- style this table in a special way maybe? -->
         <table border="1">
            <thead>
              <th>Information Content</th><th>Label</th>
            </thead>
            <tbody>
              <tr><td>[0, 0.5]  </td><td>Low      </td></tr>
              <tr><td>[0.5, 2]  </td><td>Moderate </td></tr>
              <tr><td>[2.5, 3]  </td><td>High     </td></tr>
              <tr><td>[FOO, BAR]</td><td>Very High</td></tr>
            </tbody>
         </table>
         </li>
         <li> 
         <p> Low information content indicates a very degenerate motif. While this does not have implications in terms of actual functionality, when possible, following up less degenerate motifs is more preferable.</p>
         </li>
      </ul>
    </section>
</p>


<p>
   <section>
      <b>Q16. What is a good use for atSNP DB sort order?</b>
      <ul>
        <li> 
           A16. Options for sorting the queries are useful for getting top candidates of gain or loss of function.
           For example, top loss-of-function candidates can be revealed by<br />
           Sort by:
           <p style="margin-left: 2em;"> 
              P-value Ref: Ascending<br />
              P-value Impact:  Ascending <br />
              P-value SNP: Descending<br />
           </p>
           Similarly, 
           For example, top gain-of-function candidates can be revealed by<br />
           Sort by
           <p style="margin-left: 2em;"> 
              P-value SNP: Ascending<br />
              P-value SNP Impact:  Ascending<br />
              P-value Reference: Descending<br />
           </p> 
        </li>
      </ul>
   </section>
</p>



<p>
  <section>
    <b>Q17. I would like to evaluate some SNPs that are not in atSNP DB. Any suggestions?</b>
    <ul> 
      <li>
         A17. We suggest using the <a href="https://github.com/keleslab/atSNP">atsnp R package</a>. The R package has both the JASPAR and the ENCODE motif libraries that you can evaluate your SNPs against.
      </ul>   
    </li>
   </section>
</p>


<p>
  <section>
   <b>Q18. I would like to evaluate my SNPs against PWMs that are not part of JASPAR or ENCODE. Any suggestions?</b>
     <ul> 
       <li>
          A18. We suggest using the <a href="https://github.com/keleslab/atSNP">atsnp R package</a> which allows user defined PWMs. 
       </li>
     </ul>
  </section>
</p>



<p>
  <section>
   <b>Q19. I am working on data from <a href="https://www.jax.org/strain/009376">Diversity Outbred (DO) mouse</a>. Can I use atSNP DB?</b>
   <ul> 
     <li>
        Q19. The current release of atSNP DB only has human data. We suggest using the atsnp R package for evaluating your DO SNPs. 
     </li>
   </ul>
   </section>
</p>

<!-- start debug stuff -->
{% if flavor %}
   <link href="{% static 'ss_viewer/colors/' %}{{ flavor }}.css" rel="stylesheet">
{% endif %}       
<!-- end debug stuff -->

     </div> <!-- end of content div -->
  </div>    <!-- end of container div -->

  </body>
