{% load staticfiles %}
<html>
<head>
 <meta charset="utf-8">
<title>atSNP Homepage</title>
    <link href="{% static 'ss_viewer/css/bootstrap.min.css' %}" rel="stylesheet">
    <link href="{% static 'ss_viewer/custom.css' %}" rel="stylesheet">
    <script src="{% static 'ss_viewer/js/jquery-2.2.4.js' %}"></script>
    <script src="{% static 'ss_viewer/js/bootstrap.min.js' %}"></script>

    <!-- does adding a link to the bootstrap JS make glyphicons appear? -->
</head>
<body  style="background-color: #DCDCDC;" >
<nav class="navbar navbar-default navbar-fixed-top">
<div class="container">
  <div id="content">
      {% include "ss_viewer/navbar.html" with active='home' %}  
     <img height="240" src="{% static 'ss_viewer/atsnp-logo.png' %}" />
     <p>On the front page, it would be helpful to explain the title of the
      database and state that it works exclusively for humans.</p>
<!-- todo; remove in-line styles from homepage -->
<div style="box-shadow: 1px 1px 1px 1px #888888;">
<p>

               <p><span class="glyphicon glyphicon-home"></span></p>
    @SNP DB provides statistical evaluation of impact of SNPs on transcription
    factor-DNA interactions based on position weight matrices (PWM).
    The current version is built on SNPs from dbSNP Build 144 for the
    human genome assembly GRCh38/hg38. atSNP DB uses in silico
    calculations based on the statistical method in Zuo et al.,
    Bioinformatics (2015).
    atSNP first-time users can start scanning vast amounts of
    SNP-PWM combinations with two key features:
</p>          

<ul style="list-style-type: circle;">
<li> p-values quantifying the impact of SNP on the PWM matches (p-value rank)</li>
<li> composite logo plots</li>
</ul>
<p>
   @SNP identifies and quantifies best matches to the transcription
   factor PWMs with both the reference and the SNP alleles in a small
   window around the SNP location (up to +/- 30 base pairs and considering
   subsequences spanning the SNP position). It evaluates statistical
   significance of scores on each allele and calculates statistical
   significance of the score difference between the reference and the SNP alleles.
</p>
<p>
  The discovery of best matches on the reference and SNP alleles are
   conducted separately, thus genomic subsequences providing the best matches
   with both alleles may be different. The change in goodness of match between
   both alleles indicates impact of SNPs on transcription factor binding ability
   to DNA. @SNP also provides a composite logo plot for easy visualization
   of the quality of the reference and SNP allele matches to the PWM and the
   influence of SNPs on the match change. Read more under "Help".
</p>          
</div>

<div style="box-shadow: 1px 1px 1px 1px #888888;"> 
<span>News and Updates</span>
<p>
   @SNP DB human version 1.0.0 has been released on July 20, 2016.
</p>
<p>
  The current version of @SNP enables querying 133 million SNPs from the
  <a href="http://www.ncbi.nlm.nih.gov/SNP/">NCBI database</a>
  of genetic variation (dbSNP Build 144 for human genome GRCh38)
  with 2,270 PWMs (position weight matrices) from two major sources  
  of transcription factor position weight matrix libraries
  <a href="http://jaspar.genereg.net/">JASPAR</a>
  @SNP DB contains statistical evaluations for a total of 308 billion SNP-PWM pairs.
</p>
</div>


<div style="box-shadow: 1px 1px 1px 1px #888888;">
  <span>Contact</span>
  <p>
  If you have any questions or experience technical difficulties,
  please feel free to contact us at keles@stat.wisc.edu.
  </p>
</div>




   </div>  <!-- end content div -->
</div> <!-- end container div -->
</nav>
</body>
</html>

