{% load staticfiles %}
<html>
<head>
 <meta charset="utf-8">
<title>atSNP Search Help</title>
    <link href="{% static 'ss_viewer/css/bootstrap.min.css' %}" rel="stylesheet">
    <link href="{% static 'ss_viewer/custom.css' %}" rel="stylesheet">
</head>

<body  style="background-color: #DCDCDC;" >
<nav class="navbar navbar-default navbar-fixed-top">
<div class="container">

  <div id="content">
      {% include "ss_viewer/navbar.html" with active='help' %}
      <h1>atSNP Search Help</h1>

      <a href="{% url 'ss_viewer:multi-search' %}">Back to Search</a>

      <a name="snpid"><h4>SNPid</h4> </a>
      <p>SNPid</p>

      <a name="chromosome"><h4>Chromosome</h4></a>
      <p>Chromosome </p> 

      <a name="position"><h4>Position</h4> </a>
      <p> The nucleotide position of the SNP.</p>


      <a name="ref_allele" ><h4>Reference Allele</h4></a>
      <p> Reference allele nucleotide.</p>


      <a name="snp_allele"> <h4>SNP Allele</h4></a>
      <p> SNP allele nucleotide. </p>



      <a name="pval_rank"><h4>p-value Rank</h4></a>
      <p>   
     This is the field that is restricted by the search input called "P-value cutoff". <br />
      P-values for the log rank ratio between the reference and the SNP alleles.
     </p>




      <a name="log_lik_ref"><h4>Log Likelihood Reference</h4></a>
      <p>The log-likelihood score for the reference allele.</p>



      <a name="log_lik_snp"><h4>Log Likelihood SNP</h4></a>
      <p>The log-likelihood score for the SNP allele.</p>


      <a name="log_lik_ratio"><h4>Log Likelihood Ratio</h4></a>
      <p>The log-likelihood ratio. </p>

      <a name="pval_ref"><h4>p-value Reference</h4></a>
      <p>P-values for scores on the reference allele.</p>


      <a name="pval_snp"><h4 >p-value SNP</h4></a>
      <p>P-values for scores on the SNP allele.</p>


      <a name="pval_diff"><h4>p-value Difference</h4></a>
      <p>
      P-values for the difference in scores between the reference and the SNP alleles.
      </p>


      <a name="trans_factor"> <h4>Transcription Factor</h4></a>
      <p>Transcription factor name.</p>


      <a name="motif"><h4>Motif</h4></a>
      <p>Motif name.</p>


      <a name="has_plot"><h4>Plot Available</h4></a>
      <p>
        Indicates if composite logo plot is available or not (only composite logo plots with pval_rank<0.05 are available).
      By clicking “Show plot”, the page scrolls back up to the composite logo plot of the SNP-motif pair. atSNP database has all composite logo plots with a significance change on the matches at level 0.05. Composite logo plots of SNP-motif pairs with no significant changes at level 0.05 are not available on the web. User interested in plotting them can create plots with atSNP R package available at <a href="https://github.com/keleslab/atSNP">https://github.com/keleslab/atSNP</a>. We provide two examples below to help users quickly understand how to read the plot.
      </p>

      <p>
          Plots shown are in SVG format. 
          In order to download a plot, right-click on the plot and click "Save Image As", or the equivalent on your browser.
      </p>

      <div>
          <p> 
          <div>
             <img height="320" src="{% static 'ss_viewer/example_rs572373948.png' %}"/>
          </div>
         </p>
         <p>
          <div >
             <img height="320" src="{% static 'ss_viewer/example_rs199894601.png' %}"/>
          </div>
         </p>
      </div>

      <a href="{% url 'ss_viewer:multi-search' %}">Back to Search</a>

   </div>
</div>

</body>
</nav> <!-- is this closing nav tag really needed? -->
</html>

