{% load staticfiles %}
<html>
<head>
 <meta charset="utf-8">
<title>atSNP Search Help</title>
    <script type="text/javascript"  src="{% static 'ss_viewer/jquery-2.2.4.min.js' %}"></script>
    <link href="{% static 'ss_viewer/css/bootstrap.min.css' %}" rel="stylesheet">
    <link href="{% static 'ss_viewer/custom.css' %}" rel="stylesheet">
    <script type="text/javascript" src="{% static 'ss_viewer/google_analytics.js' %}"></script>
<style type="text/css">

table.example {
 text-align:center;
 font-size:12px;
 border: 1px solid black;
}
table.example td, th { 
border: 1px solid black;
padding: 3px 20px 3px 20px; }

span.subsequence-box {
  padding: 1px;
  border: 1px solid black;
}


hgroup {
  font-weight: bold;
  color: blue;
}

table.doc-table { 
 border: 1px solid black;
 width: auto;
 margin-left: auto;
  margin-right: auto;
}
table.doc-table td, th {
 border: 1.5px solid black;
 padding-left: .4em;
 padding-right: .4em;
 padding-top: .2em;
 padding-bottom: .2em;
 text-align: center;
}
table.doc-table  th { 
 text-align:center;
}

table.borders { border: 1px solid black; }
table.borders td { border: 1px solid black;
                   padding: 5px; }


pre { 
   background-color: white;
   border: none;
   font-size: 12px;
   margin: 0px;
}
figcaption{
   font-size:14px;
   font-weight: bold;
   color: cornflowerblue;
}
figure { 
    margin: 1em;
}
cite, cite a {
    font-style: normal;
    font-size: 12px;
}
h6 {
    margin-bottom: 3px;
}
h5 {
   margin-bottom: 2px;
}
section li {
   margin-top: 1em;
}
.disc {
   list-style-type: disc;
} 
span.red {  color : red; }
td.red {    color: red; }
.underline {
   text-decoration: underline;
}

ul.refs { padding: 0px; }
/* get rid of the style above? */ 

ol.refs { padding: 0px; } 
.refs li { margin-top: 2px; }
.sequence {  font-size: 16px; 
             font-family: monospace; }
section { 
    margin-left: 2em;
    font-size: 16px;
}

/* begin style content from : 
https://www.w3schools.com/howto/tryit.asp?
       filename=tryhow_js_accordion_animate 
*/

div.accordion {
    background-color: #eee;
    cursor: pointer;
    padding-top: 10px;
    padding-bottom: 10px;
    padding-left: 1em;
    width: 100%;
    border: none;
    border-bottom: 1px solid black;

    text-align: left;
    outline: none;
    font-size: 15px;
    transition: 0.6s;
}

div.accordion.active, button.accordion:hover {
    background-color: #ddd;
}
div.panel {
    padding: 0 18px;
    background-color: white;
    max-height: 0;
    overflow: hidden;
    transition: max-height 0.2s ease-out;
}
/* end style content from source cited above */ 

.pwmlabels td{ padding: 8px; }
.pwmlabels { 
  position: relative;
  float: left;
 }
.matrix {
    position: relative;
}
body {
    padding-left: 20px;    
    padding-right: 20px;
    padding-bottom: 20px;
}
.matrix:before, .matrix:after {
    content: "";
    position: absolute;
    top: 0;
    border: 1px solid #000;
    width: 6px;
    height: 100%;
}
.matrix:before {
    left: -6px;
    border-right: 0px;
}
.matrix:after {
    right: -6px;
    border-left: 0px;
}
.matrix td {
    padding: 8px;    
    text-align: right;
}

#ic_values td.range { text-align: center; }
#ic_values th { padding: .3em;
                text-align: center; 
                max-width: 12em;
              }
#ic_values td { padding: .2em .3em .2em .4em; }

</style>

</head>

<body>
<div class="container">
  <div id="content">
      {% include "ss_viewer/navbar.html" with active='help' %}
      <h3>atSNP Search Help</h3>
     <section>  <!-- start of part 1 -->
        <div id="part-1" class="accordion">
            <hgroup><a name="part-1"><h4>1.&nbsp;Understanding motif libraries</h4></a></hgroup>
        </div>
        <div class="panel">
        <p> Motif libraries provide nucleotide patterns that transcription factors recognize well 
        from a set of transcription factor binding sites. The binding profiles are represented 
        with position weight matrices (PWMs). PWMs are used for probabilistic description of the 
        patterns in genomic sequences. The PWM is visualized as a sequence logo, where letter 
        heights are scaled based on their information content.
        </p>
        <ul class="disc">
          <li> JASPAR 2014 Core Vertebrata </li>
          <section><p><a href="http://jaspar2014.genereg.net/">JASPAR</a> is the largest open-access database of 
                   matrix-based nucleotide profiles describing the binding preferences of transcription
                   factors from multiple species (Mathelier et al., 2013). JASPAR 2014 Core Vertebrata provides
                   position frequency matrices for 205 non-redundant profiles and their summaries. The
                   position frequency matrix is converted into the position weight matrix, which carries the 
                   same information.                     </p>
                   <p>
                   <h5> <span class="underline">Example<span></h5>
                    The JASPAR library contains a position frequency matrix for a transcription factor RUNX1
                    and its sequence logo as below. Its summary is presented 
                    <a href="http://jaspar2014.genereg.net/cgi-bin/jaspar_db.pl?ID=MA0002.2&rm=present&collection=CORE"> here</a>.
                   </p>
                   <p>
                   <figure>
                   <img src="{% static 'ss_viewer/docs/image01.png' %}" height="200px"  id="figure-01"/>
                   <figcaption>
                   Figure 1 Sequence logo of MA0002.2 motif for transcription factor RUNX1
                   </figcaption>
                   </figure>

        <figure>
<div class="pwm" align="center" style="width:710px;">
<table class="pwmlabels">
<tr> <td>  A</td> </tr>              
<tr> <td>  C</td> </tr>              
<tr> <td>  G</td> </tr>              
<tr> <td>  T</td> </tr>
</table>
<table class="matrix">
  <tr>
     <td>287  </td><td>234   </td><td>123  </td><td>57   </td>
     <td>0    </td><td>87    </td><td>0    </td><td>17   </td>
     <td>10   </td><td>131   </td><td>500  </td>
  </tr> 
  </tr> 
<tr>
    <td>496</td><td>485</td><td>1072</td><td>0</td>
    <td>75 </td><td>127</td><td>0   </td><td>42</td>  
    <td>400</td><td>463</td><td>158 </td> 
</tr>

  <tr>             
     <td>696  </td><td>467</td><td>149 </td><td>7   </td>
     <td>1872 </td><td>70 </td><td>1987</td><td>1848</td>
     <td>251  </td><td>81 </td><td>289 </td>
  </tr>                                                                                                 
  <tr>  
     <td>521 </td><td>814 </td><td>656 </td><td>1936</td> 
     <td>53  </td><td>1716</td><td>13  </td><td>93  </td>
     <td>1339</td><td>1325</td><td>1053</td>
  </tr>
</table>  
 </div>
        <figcaption>PFM of MA0002.2 motif for transcription factor RUNX1</figcaption>
        </figure>
           </p>
          </section>

          <li> ENCODE Library </li>
          <section>
          The ENCODE library features systematic discovery and characterization of regulatory motifs in transcription factor binding experiments (Kheradpour et al., 2014). These PWSs are available <a href="http://compbio.mit.edu/encode-motifs/">here</a>. This collection contains known motifs from the literature 
 (TRANSFAC, JASPAR Homo Sapiens, and UniPROBE), in addition to the <i>de novo </i> motifs from hundreds of transcription factor binding experiments performed by <a href="encodeproject.org">ENCODE</a>. 
These motifs were discovered using established tools such as MDscan , MEME, Trawler, and AlignACE . 
All known and discovered motifs are grouped as in the table <a href="http://compbio.mit.edu/encode-motifs/">here</a>. 
For known ENCODE motifs, the naming convention indicates the source which initially reported the motif. For <i>de novo</i> motifs, the names contain information about the experiments ther were used to discover the motif. This information includes the name of the lab and the names of any motif discovery tools used. 
</section>

           <section> <!-- begin exapmle section -->
            <h5> <span class="underline">Example<span></h5>
            <p>ENCODE library provides 22 known motifs and 6 discovered motifs for transcription factor GATA, which are available <a href="http://compbio.mit.edu/encode-motifs/tf.html?GATA">here</a>. Here, we present one known GATA motif (GATA_known12) and one discovered GATA motif (GATA_disc1). GATA_known12 is from JASPAR Homo Sapiens and GATA_disc1 is obtained from ENCODE Snyder Lab GATA1 ChiP-seq TF experiments with the MDscan tool.</p>
            <p>
            <figure>
             <!-- filenames for images do not match up to captioned figure numbers! -->
             <img src="{% static 'ss_viewer/docs/image04.png' %}" height="200px"  id="figure-02"/>
             <figcaption>Figure 2 Sequence logo of GATA_known12 motif</figcaption>
            </figure>


            <figure>

<span align="center" > <!-- set a margin-left on this span if it needs to hover over
                            the columns of the matrix -->
>GATA_known12    GATA2_2    GATA2_jaspar_MA0036.1
</span> 
<div class="pwm" align="center" style="width:440px;">
<table class="pwmlabels">
<tr> <td>  A</td> </tr>              
<tr> <td>  C</td> </tr>              
<tr> <td>  G</td> </tr>              
<tr> <td>  T</td> </tr>
</table>
<table class="matrix">
<tr>  <td>0.245</td><td>0.000</td><td> 0.981</td><td> 0.000</td><td>0.472</td>
</tr>                                                                         
<tr>  <td>0.245</td><td>0.094</td><td> 0.000</td><td> 0.000</td><td>0.132</td>
</tr>
<tr>  <td>0.340</td><td>0.906</td><td> 0.019</td><td> 0.000</td><td>0.283</td>
</tr>
<tr>  <td>0.170</td><td>0.000</td><td> 0.000</td><td> 1.000</td><td>0.113</td>
</tr>
</table>
</div>
<figcaption>PWM of GATA_known12 motif</figcaption>
</figure>
<!-- previous version, just in case..
            <figure>
<pre>>GATA_known12    GATA2_2    GATA2_jaspar_MA0036.1
A    [0.245    0.000    0.981    0.000    0.472]
C    [0.245    0.094    0.000    0.000    0.132]
G    [0.340    0.906    0.019    0.000    0.283]
T    [0.170    0.000    0.000    1.000    0.113]</pre>
         <figcaption>PWM of GATA_known12 motif</figcaption>
         </figure> -->

         <figure>
           <img src="{% static 'ss_viewer/docs/image03.png' %}" height="200px" id="figure-03"/>
           <figcaption>Figure 3 Sequence logo of GATA_disc1 motif</figcaption>
          </figure>



<figure>
<span align="center" > <!-- set a margin-left on this span if it needs to hover over
                            the columns of the matrix -->
>GATA_disc1 GATA1_K562b_encode-Snyder_seq_hsa_r1:MDscan#1#Intergenic
</span> 
<div class="pwm" align="center" style="width:750px;">
<table class="pwmlabels">
  <tr><td>A</td></tr>              
  <tr><td>C</td></tr>              
  <tr><td>G</td></tr>              
  <tr><td>T</td></tr>
</table>
<table class="matrix">
<tr>  <td>0.323</td><td>0.138</td><td>0.667</td><td>0.000</td><td>1.000</td>     
      <td>0.000</td><td>0.878</td><td>0.974</td><td>0.000</td><td>0.243</td>  
</tr>                                                                        
<tr>  <td>0.185</td><td>0.476</td><td>0.101</td><td>0.000</td><td>0.000</td>     
      <td>0.000</td><td>0.000</td><td>0.026</td><td>0.053</td><td>0.249</td>  
</tr>                                                                        
<tr>  <td>0.360</td><td>0.323</td><td>0.000</td><td>1.000</td><td>0.000</td>     
      <td>0.000</td><td>0.000</td><td>0.000</td><td>0.947</td><td>0.508</td>  
</tr>                                                                        
<tr>  <td>0.132</td><td>0.063</td><td>0.233</td><td>0.000</td><td>0.000</td> 
      <td>1.000</td><td>0.122</td><td>0.000</td><td>0.000</td><td>0.000</td>  
</tr>
</table>
</div>
<figcaption>PWM of GATA_disc1 motif</figcaption>
</figure>
<!-- beginning of original version of this matrix -->
<!-- <figure>
<pre>
>GATA_disc1 GATA1_K562b_encode-Snyder_seq_hsa_r1:MDscan#1#Intergenic
A    [0.323    0.138    0.667    0.000    1.000    0.000    0.878    0.974    0.000    0.243]
C    [0.185    0.476    0.101    0.000    0.000    0.000    0.000    0.026    0.053    0.249]
G    [0.360    0.323    0.000    1.000    0.000    0.000    0.000    0.000    0.947    0.508]
T    [0.132    0.063    0.233    0.000    0.000    1.000    0.122    0.000    0.000    0.000]
</pre>
<figcaption>PWM of GATA_disc1 motif</figcaption>
</figure> -->
<!-- end of original version of this matrix -->
</p>
            
          <section>
          <h6><span class="underline">References</span></h6>
<ul class="refs">
<li><cite>Mathelier, A., Zhao, X., Zhang, A. W., Parcy, F., Worsley-Hunt, 
    R., Arenillas, D. J., Buchman, S., Chen, C.-y., Chou, A., Ienasescu,
    H., Lim, J., Shyr, C., Tan, G., Zhou, M., Lenhard, B., Sandelin, A. 
   and Wasserman, W. W. 
  “JASPAR 2014: an extensively expanded and updated open-access database of transcription factor binding profiles.” Nucleic Acids Research, 2013
</cite></li>
<li><cite>  Kheradpour, Pouya, and Manolis Kellis. "Systematic discovery and characterization of regulatory motifs in ENCODE TF binding experiments." Nucleic Acids Research 42.5 (2014): 2976-2987.
</cite></li><li><cite>
</cite></li><li><cite>  Matys,V., Fricke,E., Geffers,R., Gossling,E., Haubrock,M., Hehl,R., Hornischer,K., Karas,D., Kel,A.E., Kel-Margoulis,O.V. et al. (2003) TRANSFAC(R): transcriptional regulation, from patterns to profiles. Nucleic Acids Res., 31, 374–378.
</cite></li><li><cite>
</cite></li><li><cite>  Sandelin,A., Alkema,W., Engstrm,P., Wasserman,W.W. and Lenhard,B. (2004) JASPAR: an open-access database for eukaryotic transcription factor binding profiles. Nucleic Acids Res., 32, D91–D94.
</cite></li><li><cite>
</cite></li><li><cite>  Berger,M.F., Philippakis,A.A., Qureshi,A.M., He,F.S., Estep,P.W. and Bulyk,M.L. (2006) Compact, universal DNA microarrays to comprehensively determine transcription-factor binding site specificities. Nat. Biotechnol., 24, 1429–1435.
</cite></li><li><cite>
</cite></li><li><cite>  Badis,G., Berger,M.F., Philippakis,A.A., Talukder,S., Gehrke,A.R., Jaeger,S.A., Chan,E.T., Metzler,G., Vedenko,A., Chen,X. et al. (2009) Diversity and complexity in DNA recognition by transcription factors. Science, 324, 1720–1723.
</cite></li><li><cite>
</cite></li><li><cite>  Berger,M.F., Badis,G., Gehrke,A.R., Talukder,S., Philippakis,A.A., Pea-Castillo,L., Alleyne,T.M., Mnaimneh,S., Botvinnik,O.B., Chan,E.T. et al. (2008) Variation in homeodomain DNA binding revealed by high-resolution analysis of sequence preferences. Cell, 133, 1266–1276.
</cite></li><li><cite>
</cite></li><li><cite>  Jolma,A., Yan,J., Whitington,T., Toivonen,J., Nitta,K.R., Rastas,P., Morgunova,E., Enge,M., Taipale,M., Wei,G. et al. (2013) DNA-binding specificities of human transcription factors. Cell, 152, 327–339.
</cite></li><li><cite>
</cite></li><li><cite>  Hughes,J.D., Estep,P.W., Tavazoie,S. and Church,G.M. (2000) Computational identification of Cis-regulatory elements associated with groups of functionally related genes in Saccharomyces cerevisiae. J. Mol. Biol., 296, 1205–1214.
</cite></li><li><cite>
</cite></li><li><cite>  Liu,X.S., Brutlag,D.L. and Liu,J.S. (2002) An algorithm for finding protein-DNA binding sites with applications to chromatinimmunoprecipitation microarray experiments. Nat. Biotechnol., 20, 835–839.
</cite></li><li><cite>
</cite></li><li><cite>  Bailey,T.L. and Elkan,C. (1994) Fitting a mixture model by expectation maximization to discover motifs in biopolymers. Proc. Int. Conf. Int. Syst. Mol. Biol., 2, 28–36.
</cite></li><li><cite>
</cite></li><li><cite>  Pavesi,G., Mauri,G. and Pesole,G. (2001) An algorithm for finding signals of unknown length in DNA sequences. Bioinformatics, 17, S207–S214.
</cite></li><li><cite>
</cite></li><li><cite>  Ettwiller,L., Paten,B., Ramialison,M., Birney,E. and Wittbrodt,J. (2007) Trawler: de novo regulatory motif discovery pipeline for chromatin immunoprecipitation. Nat. Methods, 4, 563–565.
</cite></li>
</ul>
          </section>
          </ul> <!-- end of list containing info on JASPAR and ENCODE libraries -->

         </div> <!-- end of first accordion panel -->
        </section> <!-- end of part 1 -->
         
       <section> <!-- start of part 2 -->

        <div id="part-2" class="accordion">
           
           <hgroup>
                <h4><a name="part-2">2.&nbsp;atSNP <i>in-silico</i> Statistical Method (Zuo et al., 2015, Bioinformatics)</a></h4>
           </hgroup>
        </div>
        <div class="panel">
              <p><h3><b>OVERVIEW AND PROCEDURE</b></h3></p>
              <section><h4> <b>OVERVIEW</b> </h4>
                <figure>
                   <img height="340" src="{% static 'ss_viewer/docs/image06.png' %}" id="figure-04"/>
                   <figcaption>Figure 4 atSNP flowchart</figcaption>
                </figure>
                <p> atSNP takes as input SNP information/genomic subsequences and a PWM for a transcription factor. 
                    atSNP uses a first order Markov model and an importance sampling technique to test the statistical 
                    significance of the best match between the PWM and the subsequences overlapping the SNP position 
                    with both the reference and the SNP alleles. More imporantly, it further tests the impact of SNPs on the 
                    matches to the PWM. The testing results are summarized with a table and a graphical
                    output named composite logo plot. To learn more about the outputs, visit 
                    <a onclick="openAndScroll(true);" href="#part-3:part-3">"Interpreting search output”</a> and 
                    <a onclick="openAndScroll(true);" href="#part-4:part-4">“Reading composite logo plot”</a>.</p>


                <figure>
                   <img height="150" src="{% static 'ss_viewer/docs/image05.png' %}" id="table-01"/>
                   <figcaption>
                   Table 1 Comparison of atSNP with existing methods detecting impact of SNPs on transcription factor binding
                   </figcaption> 
                </figure>
                <p> Table 1 provides a comparison of functionality beween atSNP's statistical method and alternative methods.
                    atSNP provides the most comprehensive information on identifying impact of SNPs on transcription factor binding. 
                    Only atSNP and is-rSNP provide statistical quantification of the impact of SNPs on the binding. Furthermore, 
                    atSNP addresses the scalability issue with an efficient computation enabled by an importance sampling technique. 
                    atSNP visualizes the best matches with both the reference and the SNP alleles and their change created by the SNP.  
                    atSNP Search results are obtained by the R package <a href="https://github.com/keleslab/atSNP">atSNP</a>, that implements this procedure.
                </p>
              </section> <!-- end of atSNP overview section -->

              <section> <!-- start of atSNP Procedure section  -->
               <h4><b>PROCEDURE</b></h4>
               <ul>
                 <li><b>Step 1: affinity score                                 </b> </li>
                   <p> Using a PWM as a set of parameters for a product multinomical model, atSNP computes log likelihoods for all subsequences overlapping the SNP location that have the same length as the motif. The best matches are those with the largest log likelihood on both the SNP and reference alleles. The scores of the best matches are used for statistical testing. </p>
                   <p>
                   <h5> <span class="underline">Example<span></h5>
                   We evaluate impact of a SNP, rs117959046, on transcription factor GATA2 binding, with GATA_known12 motif:
                <figure>
                   <img src="{% static 'ss_viewer/docs/image04.png' %}" height="200px"  id="figure-05"/>
                   <figcaption> Figure 5 Sequence logo of GATA_known12 motif </figcaption> 
                </figure>
<!-- begin original version of matrix -->
<!-- <figure>
<pre>A   [0.245        0.000        0.981        0.000        0.472]
C        [0.245        0.094        0.000        0.000        0.132]
G        [0.340        0.906        0.019        0.000        0.283]
T        [0.170        0.000        0.000        1.000        0.113].</pre>
</figure> -->
<!-- end oritinal version of matrix -->
<figure>
<div class="pwm" align="center" style="width:400px;">
<table class="pwmlabels">
  <tr><td>A</td></tr>              
  <tr><td>C</td></tr>              
  <tr><td>G</td></tr>              
  <tr><td>T</td></tr>
</table>
<table class="matrix">
  <tr><td>0.245</td><td>0.000</td><td>0.981</td><td>0.000</td><td>0.472</td></tr> 
  <tr><td>0.245</td><td>0.094</td><td>0.000</td><td>0.000</td><td>0.132</td></tr>
  <tr><td>0.340</td><td>0.906</td><td>0.019</td><td>0.000</td><td>0.283</td></tr>
  <tr><td>0.170</td><td>0.000</td><td>0.000</td><td>1.000</td><td>0.113</td></tr>
</table>
</div>
</figure>

               </p>
               <p>
The motif length is 5 and sequences of length 9 centered at the SNP position are extracted from both the reference allele and the SNP variation of the reference allele:
               </p>
               <p>
               <table class="borders"><tbody>
                <tr> <td style="padding-right: 8px;" >Reference allele:</td>
                     <td  class="sequence" >AAGA<span class="red">C</span>AAGC</td></tr>
                <tr> <td style="padding-right: 8px;">SNP allele:       </td>
                     <td  class="sequence"  >AAGA<span class="red">T</span>AAGC</td></tr>
               </tbody></table>
               </p>
               <p> Next, we compute log likelihoods of all subsequences of length 5 spanning the SNP position and find the best match, which is associated with the largest log likelihood. The second subsequence on the forward strand (A<span class="subsequence-box">AGA<span class="red">C</span>A</span>CG) has the largest log likelihood with the reference allele. (Table 2). The second subsequence on the forward strand also has the largest log likelihood with the SNP allele (Table 3).</p>
                 <p>
                <figure>
                   <table class="example table-striped">
                          <thead><tr> <th>Reference Allele     </th> 
                                  <th>Forward strand score </th>
                                  <th>Reverse strand score </th>  </tr>
                         </thead >
                   <tbody>
                    <tr><td class="sequence" ><span class="subsequence-box">AAGA<span class="red">C</span></span>AAGC</td> <td> -53.45</td> <td> -49.31</td></tr>
                    <tr><td class="sequence" >A<span class="subsequence-box">AGA<span class="red">C</span>A</span>AGC</td> <td class="red"> -25.30</td> <td> -50.10</tr></td>
                    <tr><td class="sequence" >AA<span class="subsequence-box">GA<span class="red">C</span>AA</span>GC</td> <td> -70.91</td> <td> -30.79</tr></td>
                    <tr><td class="sequence" >AAG<span class="subsequence-box">A<span class="red">C</span>AAG</span>C</td> <td> -28.08</td> <td> -72.66</tr></td>
                    <tr><td class="sequence" >AAGA<span class="subsequence-box"><span class="red">C</span>AAGC</span></td> <td> -49.50</td> <td> -27.73</tr></td>
                   </tbody></table>
                   <figcaption> Table 2 Log likelihoods of all subsequences with the reference allele</figcaption> 
                </figure>
                <figure>
                   <table class="example table-striped">
                          <thead><tr> <th>SNP Allele       </th> 
                                      <th>Forward strand score </th>
                                      <th>Reverse strand score </th> </tr></thead >
                   <tbody>
                  <tr> <td class="sequence"> <span class="subsequence-box">AAGA<span class="red">T</span></span>AAGC  </td> <td> -53.60 </td> <td> -49.64</td> </tr>
                  <tr> <td class="sequence"> A<span class="subsequence-box">AGA<span class="red">T</span>A</span>AGC  </td> <td class="red"> -2.28  </td> <td> -73.03</td> </tr>
                  <tr> <td class="sequence"> AA<span class="subsequence-box">GA<span class="red">T</span>AA</span>GC  </td> <td> -70.91 </td> <td> -26.84</td> </tr>
                  <tr> <td class="sequence"> AAG<span class="subsequence-box">A<span class="red">T</span>AAG</span>C  </td> <td> -48.74 </td> <td> -72.66</td> </tr>
                  <tr> <td class="sequence"> AAGA<span class="subsequence-box"><span class="red">T</span>AAGC</span>  </td> <td> -49.87 </td> <td> -27.22</td> </tr>
                   </tbody></table>
                   <figcaption>Figure 6 Log likelihoods of all subsequences with the SNP allele</figcaption> 
                </figure>
                 </p><!-- end paragraph of figures -->
                 <li><b>Step 2: p-value for the affinity score                 </b> </li>
                 <p>The p-value for the affinity score with the reference allele (p-value Reference) is the probability that a random sequence has an affinity score larger than or equal to the best match score with the reference allele. atSNP simulates random sequences using a first order Markov model. It estimates the p-value as the frequency of randomly getting as large or larger affinity scores than the best match score with the reference allele. Similarly, the p-value for the affinity score with the SNP allele (p-value SNP) is estimated as the frequency of getting random scores as large or larger than the best match score with the SNP allele.</p>
                   <p>
                   <h5> <span class="underline">Example (Continued)<span></h5>
(Continued From Step 1) The log likelihood score for the reference allele is <span class="red">-25.30</span>, and the log likelihood score for the SNP allele is <span class="red">-2.28</span>. The p-value for the reference allele is the probability that a random sequence has an affinity score larger than or equal to <span class="red">-25.30</span>, which is our target region (Figure 7). For such a small p-value, naive Monte-Carlo generates a small number of scores in the target region (Figure 8), thus requires a large number of simulations. Alternatively, in atSNP, the importance sampling generates many more scores in the target region (Figure 9). The p-value for the reference allele is then estimated by the weighted frequency as p-value Reference = <span class="red">0.499</span>. The importance sampling algorithm significantly reduces the required number of simulations. Similarly, the p-value for the SNP allele, p-value SNP, is estimated as <span class="red">0.009</span>. 
                   </p>
                   <p>
                <figure style="float: left;">
                   <img height="260" src="{% static 'ss_viewer/docs/image13.png' %}" id="figure-07"/>
                   <figcaption>Figure 7 Distribution of scores <br />
                               from random motif sequences.</figcaption> 
                </figure>
                <figure style="float:left;">
                   <img height="260" src="{% static 'ss_viewer/docs/image11.png' %}" id="figure-08"/>
                   <figcaption> Figure 8 Empirical distribution<br />
                                of simulated random scores with the naive <br />
                                Monte Carlo method.
                  </figcaption> 
                </figure>
                <figure>
                   <img height="260" src="{% static 'ss_viewer/docs/image00.png' %}" id="figure-09"/>
                   <figcaption> Figure 9 Empirical distribution of simulated <br />
                                random scores with importance sampling.
                  </figcaption> 
                </figure>
                   <p style="clear: both;">
                   </p>
                  </p> 
                 <li><b>Step 3: p-value for the affinity change between alleles</b> </li>
                   <p> atSNP tests the significance of <i>log rank ratio</i>: log(p-value Reference)-log(p-value SNP). atSNP computes the p-value based on a Monte-Carlo simulation that uses both the reference and SNP alleles together. This is done using the importance sampling technique as described in <b>Step 2</b>.</p>
                   <p>
                   <h5> <span class="underline">Example (Continued)<span></h5> 
The log rank ratio for the impact of SNP, rs117959046, is <span class="red">4.02</span>. With the importance sampling, the p-value for the log rank ratio is estimated as p-value rank = <span class="red">0.005</span>. In atSNP Search, this p-value is referred to as p-value SNP Impact.</p>

              <section> <!-- begin Reference section -->
                <p>
                <h6><span class="underline">Reference</span></h6>
                 <ul class="refs">
                    <li><cite> Zuo, Chandler, Shin, Sunyoung, and Keleş, Sündüz. atSNP: transcription factor binding affinity testing for regulatory SNP detection. Bioinformatics (2015): 31, 3353-3355
                    </cite></li>
                    <li><cite> Andersen, Malin C., Engstrom, Par G., Lithwick, Stuart, Arenillas, David, Eriksson, Per, Lenhard, Boris, Wasserman, Wyeth W., and Odeberga, Jacob. In silico detection of sequence variations modifying transcriptional regulation. PLoS Computational Biology 4.1 (2008): e5.
                   </cite></li>
                   <li><cite> Grant, Charles E., Timothy L. Bailey, and William Stafford Noble. FIMO: scanning for occurrences of a given motif. Bioinformatics 27.7 (2011): 1017-1018.
                   </cite></li>
                   <li><cite> Macintyre, Geoff, Bailey, James, Haviv, Izhak, and Kowalczyk, Adam. is-rSNP: a novel technique for in silico regulatory SNP detection. Bioinformatics 26.18 (2010): i524-i530.
                   </cite></li>
                   <li><cite> Riva, Alberto. Large-scale computational identification of regulatory SNPs with rSNP-MAPPER. BMC genomics 13.4 (2012): S7.
                   </cite></li>
                  <li><cite> Thomas-Chollier, Morgane, Hufton, Andrew, Heinig, Matthias, O'keeffe, Sean, Masri, Nassim E., Roider, Helge G., Manke, Thomas, and Vingron, Martin. Transcription factor binding predictions using TRAP for the analysis of ChIP-seq data and regulatory SNPs. Nature protocols 6.12 (2011): 1860-1869.
                  </cite></li>
                 </ul>
               </section> <!-- end of Reference section (inside procedure section)  -->
               </ul>
               </p>
           </section><!-- end of atSNP Procedure section  -->
         </div> <!-- end of accordion panel for part 2 -->
         </section> <!-- end of part 2 -->



         <section> <!-- start of part 3 -->

        <div id="part-3" class="accordion">
           <hgroup><a name="part-3"><h4>3.&nbsp;Interpreting Search Output</h4></a></hgroup>
        </div>
        <div class="panel">
           <p>atSNP search output is grouped into four general categories: SNP information, motif information, atSNP scores, and atSNP p-values.
              Details for these categories are summarized in the following tables:</p>
          <section>  <!--begin section full of tables --> 
           <p>            
            <table class="doc-table" id="snpTable">
            <thead><tr><th colspan="2">SNP information</th></tr></thead>
            <tbody>
            <tr>
              <td style="width:18em;"> RS ID (<b>R</b>eference<b>S</b>NP      
                   <b>ID</b> number)                                          </td> 
              <td> Identification tag assigned by NCBI to a group of SNPs that 
                   map to an identical location<sup>1</sup> A link to the DBSNP page for any SNP shown in the data is available by clicking on the SNPid. 
              </td> 
            </tr>
            <tr>
              <td> Coordinate                                                 </td> 
              <td> Formatted as chromosome:position. One of the human genome (assembly GRch38/hg38) chromosomes, e.g. Chr1-Chr22, ChrX, ChrY, ChrM.
                   The position is the 1-based location of the SNP relative to the first base at the 5’ 
                   end of the forward strand of the reference 
                   sequence <sup>2</sup>. A link to data around the genomic coordinate in the UCSC genome browser is available by clicking on the coordinate.   
              </td>
            </tr>
            <tr>
              <td> Reference and SNP Alleles                                  </td> 
              <td> Formatted as Reference allele/SNP allele.  e.g.: T/G.
                                                                              </td>
            </tr>
            <tr>
              <td> Reference and SNP Strands                                  </td>
              <td> The reference and SNP strand information is shown on the composite logo plot. 
                   It indicates whether the displayed sequences are in the forward or reverse direction. 
                                                                              </td>
            </tr>
            </tbody>
            </table> 
            <ol>
            <li><a href="http://www.ncbi.nlm.nih.gov/books/NBK44417/#Content.what_is_a_reference_snp_or__rs_i">
                     http://www.ncbi.nlm.nih.gov/books/NBK44417/#Content.what_is_a_reference_snp_or__rs_i</a></li>
            <li><a href="https://bioconductor.org/packages/release/data/annotation/manuals/SNPlocs.Hsapiens.dbSNP144.GRCh38/man/SNPlocs.Hsapiens.dbSNP144.GRCh38.pdf">https://bioconductor.org/packages/release/data/annotation/manuals/SNPlocs.Hsapiens.dbSNP144.GRCh38/man/SNPlocs.Hsapiens.dbSNP144.GRCh38.pdf</a></li>
            <ol>
            </p>
            <p>
            <table class="doc-table" id="motifTable">                                  
            <thead><tr><th colspan="2">Motif information</th></tr></thead>
            <tbody>
              <tr>
                <td> Motif                                  </td> 
                <td> Motif name, e.g., MA0002.2 For JASPAR motifs, a link to the motif's entry in JASPAR database by clicking on the motif.              
                                                            </td> 
              </tr>
              <tr>
                <td style="width:18em;"> 
                     Transcription Factor                   </td> 
                <td> Transcription factor name, e.g., RUNX1 
                     When available, a link to the FACTORBOOK page for the transcription factor is provided by clicking on the transcription factor. 
                                                            </td> 
              </tr>
            </tbody>
            </table>                                                                    
            </p>
            <p>
            <table class="doc-table" id="scoresTable" >
              <thead><tr><th colspan="2">atSNP Log Likelihoods (evaluated wrt PWM)</th></tr></thead>
              <tbody>
                <tr>
                  <td style="width:18em;"> Rank Ratio (SNP Impact)    </td> 
                  <td> Log of ratio of p-value Reference to p-value SNP.
                       This is the main statistic that the atSNP p-values are based on. 
                  </td> 
                </tr>
                <tr>
                  <td style="width:18em;"> Reference                  </td> 
                  <td> The log likelihood score of the best match (highest scoring subsequence)
                       with the reference allele                      </td> 
                </tr>
                <tr>
                  <td>  SNP                                           </td> 
                  <td> The log likelihood score of the best match (highest scoring subsequence) 
                       with the SNP allele                            </td> 
                </tr>
                <tr>
                  <td>   Ratio                                        </td> 
                  <td> The difference between Log Likelihood Reference
                       and Log Likelihood SNP                         </td> 
                </tr>
                <tr>
                  <td style="width:18em;"> Enhance Odds           
                  </td> 
                  <td> 
                       Once atSNP identifies the best matching subsequence to the PWM with the reference allele,
                       it evaluates the likelihood of that subsequence with the reference allele and the SNP allele.
                       This quantity is the log of the ratio of these two likelihoods.
                  </td>
                </tr>
                <tr>
                  <td> Reduce Odds                                 
                  </td> 
                  <td>
                       Once atSNP identifies the best matching subsequence to the PWM with the SNP allele, it evaluates the
                       likelihood of that subsequence with the reference allele and the SNP allele. This quantity is the log 
                       of the ratio of these two likelihoods.
                  </td>
                </tr>
                </tbody>
            </table>
            </p>

            <p>
            <table class="doc-table" id="pvalueTable">
            <thead><tr><th colspan="2">atSNP p-values</th></tr></thead>
            <tbody>
            <tr>
              <td style="width:18em;"> p-value SNP Impact 
              </td> 
              <td> A significant p-value, e.g. 0.0001, statistically 
                   supports the potential gain or loss of function of
                   the genomic region with the SNP in terms of transcription 
                   factor binding.
              </td>
            </tr>
            <tr>
              <td> p-value Reference 
              </td>
              <td> p-value for score with the reference allele 
                   (Log Likelihood Reference). A significant p-value
                   indicates that the match to the PWM with the reference
                   allele is statistically supported.                                   
              </td>
            </tr>
            <tr>
              <td> p-value SNP                
              </td>
              <td> p-value for scores wth the SNP allele 
                 (Log Likelihood SNP). A significant p-value indicates
                 that the match to the PWM with the SNP allele is 
                 statistically supported.      
              </td>
            </tr>
            <tr>
              <td> <sup>*</sup>p-value Difference 
              </td>
              <td> p-value for the difference in scores with 
                   the reference and the SNP alleles (Log Likelihood
                   Ratio). This is essentially the result from a likelihood
                   ratio test. While we report this for completeness, the
                   final atSNP results are based on p-value SNP impact
                   which quantifies the significace of the 
                   <a href="#scoresTable">Log Rank Ratio</a>.
              </td> 
            </tr>
            <tr>
              <td> <sup>*</sup>p-value Condition Ref                        
                                          </td>   
              <td> Conditional p-value for scores on the reference 
                   allele based on Log Enhance Odds             </td> 
            </tr>
            <tr>
              <td> <sup>*</sup>p-value Condition SNP                    
                                          </td> 
              <td> Conditional p-value for scores on the SNP allele
                   based on Log Reduce Odds                     </td>
            </tr>
            </tbody>
            </table>
            <span><sup>*</sup> 
                While these p-values are reported for completeness, they are not used for querying.
            </span>
            </p>
            </section> <!-- end section full of tables -->
            <section> <!-- start of example section -->
            <p> <h5> <span class="underline">Example<span></h5> 
               We evaluate the impact of the SNP rs117959046 on the binding of the
               transcription factor GATA2, using the GATA_known12 motif: 
            </p>

             <p> 
             <figure>
                <img src="{% static 'ss_viewer/docs/image04.png' %}" height="200px"  id="figure-10"/>
                <figcaption>Figure 10 Sequence logo of GATA_known12 motif</figcaption> 
             </figure>
          <div class="pwm" align="center" style="width:400px;">
            <table class="pwmlabels">
              <tr><td>A</td></tr>              
              <tr><td>C</td></tr>              
              <tr><td>G</td></tr>              
              <tr><td>T</td></tr>
            </table>
            <table class="matrix">
              <tr><td>0.245</td><td>0.000</td><td>0.981</td><td>0.000</td><td>0.472</td></tr> 
              <tr><td>0.245</td><td>0.094</td><td>0.000</td><td>0.000</td><td>0.132</td></tr>
              <tr><td>0.340</td><td>0.906</td><td>0.019</td><td>0.000</td><td>0.283</td></tr>
              <tr><td>0.170</td><td>0.000</td><td>0.000</td><td>1.000</td><td>0.113</td></tr>
            </table>
          </div>
          <p>
            <table>
              <tr>
                <td style="padding: 5px;" >
                  <table class="doc-table">
                    <thead>
                      <tr><th colspan="2"> SNP Information </th></tr>
                    </thead>
                    <tbody>
                      <tr><td>rsID                </td> <td>rs117959046     </td> </tr>
                      <tr><td>Coordinate          </td> <td>ch11:134339671  </td> </tr>
                      <tr><td>Reference/SNP allele</td> <td>C/T             </td> </tr>
                    </tbody>
                  </table>
                 </td>
                 <td>
                  <table class="doc-table">
                    <thead>
                      <tr><th colspan="2">Motif Information</th></tr></thead>
                    <tbody>
                      <tr><td>Motif               </td><td>GATA2_2</td></tr>
                      <tr><td>Transcription Factor</td><td>GATA   </td></tr>
                    </tbody>
                  </table>
                 </td>
              <tr>
            </table>
          </p>
          <div style="width: 350px; margin-left: 9em; position:relative;">
            <p>
              <table class="doc-table">
                <thead>
                  <tr><th colspan="3">atSNP Results </th></tr>
                </thead>
                <tbody>
                  <tr><td>                   </td><td>Log Likelihood   </td> <td>p-value </td>  </tr>
                  <tr><td>SNP Impact (Rank Ratio)</td><td>4.02             </td> <td>0.00484 </td>  </tr>
                  <tr><td>Reference          </td><td>-25.3            </td> <td>0.499   </td>  </tr>
                  <tr><td>SNP                </td><td>-2.27            </td> <td>0.00898 </td>  </tr>
                  <tr><td>Ratio              </td><td>-23.0            </td> <td>0.0207  </td>  </tr>
                  <tr><td>Enhance Odds       </td><td>23.0             </td> <td>0.499   </td>  </tr>
                  <tr><td>Reduce Odds        </td><td>-23.0            </td> <td>0.00184 </td>  </tr>
                </tbody>
              </table>
            </p>
          </div>
          <figure>
             <img height="450" src="{% static 'ss_viewer/docs/image14.jpg' %}"/>
             <figcaption> Figure 11 Composite sequence logo for this result </figcaption> 
          </figure>
          <p>
The SNP “rs117959046” is a genetic mutation from C to T on the forward strand in Chromosome 1: 134339671. This creates a highly likely binding site for the transcription factor GATA2. This is supported by the p-value SNP Impact statistic, 0.00484. This p-value indicates significance of the binding creation by the SNP. Further, the p-value for the best match between the GATA 2 motif and the SNP allele is 0.00898, whereas p-value for the best match between the motif and the reference allele is 0.499. This implies that binding of GATA 2 transcription factor on the SNP allele is statistically significant while the significance of binding of GATA 2 on the reference allele is not statistically supported. Overall, atSNP statistically supports that the SNP significantly enhances binding for GATA 2.
          </p>
          </section> <!-- end of example section -->
         </div> <!-- end of accordion panel for section 3 -->
         </section><!-- end of part 3 -->  

         <section> <!-- start of part 4 -->
         <div id="part-4" class="accordion">
           <hgroup><a name="part-4"><h4>4.&nbsp;Reading a Composite Logo Plot</h4></a></hgroup>
         </div>
         <div class="panel">
           <p> atSNP provides a composite logo plot to visualize the impacts of SNPs on transcription factor binding. 
               These plots reveal whether the SNP might lead to gain of function or loss of function.
           </p> 
            <section> <!-- begin examples section -->
              <h5> <span class="underline">Examples<span></h5> 
             <figure>
                <img height="350" src="{% static 'ss_viewer/docs/image08.jpg' %}" id="figure-10"/>
                <figcaption>Figure 12 GATA2 motif scan for rs117959046</figcaption> 
             </figure>
              <p> 
                The SNP, rs117959046, is highly likely to create a binding site for the transcription factor GATA2 on the genome. 
                The composite logo plot illustrates this with the comparison between the best matches of GATA2_2 PWM on the reference and SNP alleles. 
                Among subsequences of length 5 overlapping the SNP position with the reference allele, “AGACA” on the forward strand has the largest log likelihood score.
                The subsequence is referred to as the “best match with the reference allele”. This subsequence is depicted on the second row with the corresponding PWM 
                on the first row. Similarly, among subsequences of length 5 overlapping the SNP position with the SNP allele, “AGATA” on the forward strand is referred
                to as the “best match with the SNP allele”. It is depicted on the third row with the corresponding PWM on the fourth row.
              </p>
             <figure>
                <img height="350" src="{% static 'ss_viewer/docs/image09.jpg' %}" id="figure-11"/>
                <figcaption>Figure 13 RUNX1 motif scan for rs199894601</figcaption> 
             </figure>
             <p>The SNP, rs199894601, is highly likely to create a binding site for the transcription factor RUNX1 on the genome. The best match position on the SNP allele is different from that on the reference allele. Specifically, the best matches occur on different strands and genomic coordinates (bp). The composite logo plot illustrates this with the comparison between the best matches of RUNX1 PWM from JASPAR on the reference and SNP alleles.
             </p>

             <p>
The beginning of the best match on the SNP allele is 3 bases before the SNP position in the 5' to 3' direction on the reverse strand.
Among subsequences of length 11 overlapping the SNP position on the SNP allele, this subsequence has the largest log likelihood score. This subsequence is referred to as “best match with the SNP allele” and depicted on the third row with the corresponding PWM on the fourth row.  Similarly, we can find “best match with the reference allele” and depict it on the second row with the corresponding PWM on the first row. The beginning of the best match on the reference allele is one base before the SNP position in the 5' to 3' direction on the forward strand.
             </p>
            </section>           <!-- end examples section -->
           </div> <!-- end of accordion panel for part 4-->
         </section> <!-- end of part 4 -->


         <section> <!-- start of part 5 -->
           <div id="part-5" class="accordion"> 
           <hgroup><a name="part-5"><h4>5.&nbsp;atSNP Search Infrastructure</h4></a></hgroup>
           </div>
           <div class="panel">
             <figure>
               <img height="770" src="{% static 'ss_viewer/atsnp-site-design.png' %}" id="figure-12" />
             </figure>
             <p> atSNP Search utilizes a scale-out systems architecture to store and retrieve approximately thirty seven billion records.  Elasticsearch, a NoSQL database, utilizes a distributed scale-out system design for large workloads. It accomplishes this task by distributing requests for searches amongst the scaled computing resources. An API handles communication between the search page and the Elasticsearch data store.
            </p>
           </div> <!-- end of accordion panel for part 5 -->
        </section> <!-- end of part 5 -->

        <section> <!-- start of part 6 -->
           <div id="part-6" class="accordion" >
           <hgroup><a name="part-6"><h4>6.&nbsp;Using atSNP Query Modifiers</h4></a></hgroup>
           </div>
           <div class="panel">
           <p>
           <h5><span><b>Swapping P-Value Cutoff Operators</b></span></h5> 
           The operator for the SNP and reference p-value cutoffs can be swapped
           between &le; and &gt;.
           To reverse the operator for one of these cutoffs, click on the button 
           with the operator on it beside the corresponding p-value.
           </p>
           <p>
           <h5><span><b>Adjusting the Sort Order</b></span></h5> 
           The sort control indicates the priority of sort fields and the
           direction of each sort field. If a field has top sort priority, then
           that field will be the one that results are sorted by. Any results 
           which have the same value for the first sort field will be ordered
           by the values of the fields with the second sort priority and so on.
           <br/> 
           To reverse the sort order for one sort field from ascending to
           descending or vice-versa, click the icon by that search field.
           To advance or demote a sort field's priority, select it (by 
           clicking on it) and click the up or down arrow.
           </p>

           <h5><span><b>Filtering by Motif Degeneracy</b></span></h5>
           This control is based on computing information content (IC) of each individual position within a PWM,
           then taking the median of the information content across all positions. 
           These median IC values are then percentiled  (25%, 50%, and 75%) to classify each PWM into the following four classes. 
             <!-- style this table in a special way maybe? -->
             <table id="ic_values" border="1">
                <thead>
                  <th>Median Information <span style="display: table; margin: auto;">Content</span>  </th>
                  <th>Motif Degeneracy <span style="display: table; margin: auto;">Label</span></th>
                </thead>
                <tbody>
                  <tr><td class="range">[0, 0.70)    </td><td>Very High  </td></tr>
                  <tr><td class="range">[0.7, 1.1)   </td><td>High       </td></tr>
                  <tr><td class="range">[1.1, 1.52)  </td><td>Moderate   </td></tr>
                  <tr><td class="range">[1.52, 2]    </td><td>Low        </td></tr>
                </tbody>
             </table>
             <p>
                 Low information content indicates a very degenerate motif. 
                 While this does not have implications in terms of actual functionality,
                 when possible, following up less degenerate motifs is more preferable.
             </p>




           <p><h5><span><b>Limitations on Numeric Queries</b></span></h5>
           Queries are numerically limited in the following ways:
<ul>
<li><h5><b>SNPid</b></h5> Up to 1,000 SNPids may be entered. 
   Plain text files or direct input are both acceptable.</li>
<li><h5><b>SNPid Window</b></h5> The window size is limited to 125 million bases. </li>
<li><h5><b>Genomic Location</b></h5> 
     The maximum region size is 250 million bases.
     The region size is calculated by subtracting the start location from the 
     end location on the selected chromosome. </li>
<li><h5><b>Gene</b></h5> The region around the gene is limited to 125 million bases. </li>
</ul>


           </p>
          </div> <!-- end of accordion panel for part 6 -->
        </section> <!-- end of part 6 -->

        <section> <!-- start of part 7 -->
           <div id="part-7" class="accordion" >
           <hgroup><a name="part-7"><h4>7.&nbsp;How to Cite</h4></a></hgroup>
           </div>
           <div class="panel">
           <p>
           <ol class="refs">
             <li><cite><a href="https://doi.org/10.1093/bioinformatics/btv328"> Zuo, Chandler, Shin, Sunyoung, and Keleş, Sündüz. atSNP: transcription factor binding affinity testing for regulatory SNP detection. Bioinformatics (2015): 31, 3353-3355</a>.
             </cite> </li>
             <li><cite>Shin, Sunyoung, Hudson, Rebecca, Harrison, Christopher, Craven, Mark, and Keleş, Sündüz. atSNP Search: a web resource for quantifying impact of dbSNP SNPs on transcription factor binding. (2018). In preparation.
             </cite> </li>
           </ol>
           </p>
          </div> <!-- end of accordion panel for part 6 -->
        </section> <!-- end of part 6 -->

        <!-- </ol> --><!-- end of ordered list of main help items -->

     </div> <!-- end of content div -->
  </div>    <!-- end of container div -->

  </body>
      <script>
      function openAndScroll(internal){
              console.log("location:");
              console.log(location);
              //console.log("event: ");
              //console.log(event);
              var hashToUse;
              var event; 
              if ( ! ( event === undefined )){ 
                  console.log("event.srcElement.hash");
                  console.log(event.srcElement.hash);
                  hashToUse =  event.srcElement.hash;
              }else{ 
                  hashToUse = location.hash;  
              }
 
              //if (!(location.hash === '')){
              if (!(hashToUse === '')){
                  console.log("there's hash in here! :" +  hashToUse);

                  var selector = hashToUse.replace('#', '');
                  var tableToScrollTo = null;

                  //if there's a colon, split on the colon...
                  if ( selector.indexOf(':') == -1 ){
                      selector = "a[name='" + selector + "']";
                  }else{
                      var parts = selector.split(':');
                      selector = "a[name='" + parts[0] + "']";
                      tableToScrollTo = parts[1];
                      //console.log("going to try to scroll to : " + tableToScrollTo);
                  } 

                  //motifTable
                  var headerlink = $(selector)[0];
                  var accdiv = headerlink.parentElement.parentElement;
                  console.log("got a selector for the accordion div");
                  console.log(accdiv);
                  accdiv.classList.toggle('active');
                  var panel = accdiv.nextElementSibling;
                  if ((panel.style.maxHeight)&&(internal === false)){   //where to check if it's internal or not.
                    panel.style.maxHeight = null;
                  } else {
                    panel.style.maxHeight = panel.scrollHeight + "px";
                  } 

                 /*http://stackoverflow.com/questions/18103534/scroll-to-specific-div-on-page-load
                   Adapted from example found on S.O.*/ 
                 if ( ! ( tableToScrollTo === null) ){
                     console.log("tableToScrollTo");

                     console.log( tableToScrollTo );
                     console.log($("#" + tableToScrollTo) );
                     console.log("tableToScrollTo.offset()");
                     console.log($("#" + tableToScrollTo).offset() );
                     // Handler for .ready() called.
                     $('html, body').animate({
                         scrollTop: $('#' + tableToScrollTo).offset().top
                     }, 'slow'); 
                 }
              }
       console.log("completed running openAndScroll");
       }//end function openAndScroll.
       openAndScroll(false); 
       console.log("running openAndScroll for docuemnt.ready");
              var acc = document.getElementsByClassName("accordion");
              var i;

              for (i = 0; i < acc.length; i++) {
                acc[i].onclick = function() {
                  this.classList.toggle("active");
                  var panel = this.nextElementSibling;
                  if (panel.style.maxHeight){
                    panel.style.maxHeight = null;
                  } else {
                    panel.style.maxHeight = panel.scrollHeight + "px";
                  } 
                }
              }
       
      </script>
